### PR TITLE
Reland "Freeze all statuses emitted by calls into dynamic modules"

### PR DESCRIPTION
**Pull request by qedawkins**
Drops the check for the filename.

ci-extra: build_test_all_windows,build_test_all_macos_arm64,build_test_all_macos_x86_64

### CODE REVIEW

File: samples/custom_module/dynamic/test/example.mlir [hunk 1]
```diff
@@ -41,4 +52,13 @@ module @example {
 
     return
   }
+
+  // CERROR-LABEL: EXEC @error
+  func.func @error() {
+    // Show an example of dumping a custom error message.
+    // CERROR-NEXT: UNKNOWN; Forced failure; while invoking C++ function custom.error;
```
Comment (benvanik) [inline samples/custom_module/dynamic/test/example.mlir]: you just want `UNKNOWN` - strings are optional and only provided if IREE_STATUS_FEATURE_ANNOTATIONS is set

Comment (qedawkins) [reply samples/custom_module/dynamic/test/example.mlir]: darn

**Review by ScottTodd — APPROVED** [review]
nit: when reverting and then re-landing changes, reviewing is easier if you use `git revert` then apply fixes in separate commits. That way, reviewers can view just the changes from the original PR. You can still squash when merging the PR.

**Comment by qedawkins** [comment]
I see, makes sense. I normally do that with regular review processes but didn't think to do that here. Thanks for the tip!

**Review by benvanik — APPROVED** [review]

**Comment by ScottTodd** [comment]
> I see, makes sense. I normally do that with regular review processes but didn't think to do that here. Thanks for the tip!

Here's an example: https://github.com/openxla/iree/pull/15267
Two commits:
image
The fixes are isolated:
image

**Pull request merged**